src/main/java/com/github/checkstyle/generatepatchfile/GeneratePatchFile.java

```diff
@@ -0,0 +1,238 @@
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.diff.DiffEntry;
+import org.eclipse.jgit.diff.DiffFormatter;
+import org.eclipse.jgit.lib.ObjectReader;
+import org.eclipse.jgit.lib.Repository;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
+import org.eclipse.jgit.treewalk.CanonicalTreeParser;
+
+/**
+ * To generate patch file through jgit.
+ */
+public class GeneratePatchFile {
+    /**
+     * Git instance.
+     */
+    private Git git;
+
+    /**
+     * RepoPath of repository that will be checked by checkstyle.
+     */
+    private String repoPath;
+
+    /**
+     * Repository instance.
+     */
+    private Repository repository;
+
+    /**
+     * Path of checkstyle-tester.
+     */
+    private String testerPath;
+
+    /**
+     * Path of checkstyle.
+     */
+    private String checkstyleRepoPath;
+
+    /**
+     * Patch-branch of checkstyle.
+     */
+    private String checkstyleBranch;
+
+    /**
+     * BaseConfigFile.
+     */
+    private File baseConfigFile;
+
+    /**
+     * PatchConfigFile.
+     */
+    private File patchConfigFile;
+
+    /**
+     * Init.
+     */
+    public GeneratePatchFile(String repoPath, String testerPath,
+                             String checkstyleRepoPath, String checkstyleBranch,
+                             File baseConfigFile, File patchConfigFile) {
+        this.testerPath = testerPath;
+        this.checkstyleRepoPath = checkstyleRepoPath;
+        this.checkstyleBranch = checkstyleBranch;
+        this.baseConfigFile = baseConfigFile;
+        this.patchConfigFile = patchConfigFile;
+        this.repoPath = repoPath;
+        final File gitDir = new File(repoPath, ".git");
+        try {
+            final Repository repository = new FileRepositoryBuilder().setGitDir(gitDir)
+                    .readEnvironment().findGitDir().build();
+            this.repository = repository;
+            git = new Git(repository);
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+    }
+
+    private List<RevCommit> getAllCommits() throws Exception {
+        Iterable<RevCommit> commits = git.log().add(repository.resolve("HEAD")).call();
+        int count = 0;
+        List<RevCommit> commitList = new ArrayList<>();
+        for (RevCommit commit : commits) {
+            System.out.println("LogCommit: " + commit +
+                    ", name: " + commit.getName() + ", id: " + commit.getId().getName());
+            count++;
+            commitList.add(commit);
+        }
+        System.out.println(count);
+        return commitList;
+    }
+
+    private void generateTwoCommitDiffPatch(RevCommit commitOld, RevCommit commitNew) throws Exception {
```

Comment:
please stop any activity to re-implement `git format-patch --stdout HEAD~2`  we will test only git and only on linux and we do not need java implementation of it during POC, probably will not need it ever.
You need to just execute shell command `git format-patch --stdout HEAD~1` , `git format-patch --stdout HEAD~2` ,..... to get patch file. To generate patch for two commits in one file `git format-patch -2 --stdout`

Please keep this implementation as is and make implementations for usage of shell execution of git commands.